src/actions.js

```diff
@@ -127,6 +127,17 @@ export function fetchLocationsStr(mm, level, parent, str, first) {
   return graphql(payload, `LOCATION_LOCATIONS_${level}`);
 }
 
+export function fetchParentLocationsStr(mm, level, parentUuids, str, first) {
+  const types = mm.getConf("fe-location", "Location.types", ["R", "D", "W", "V"]);
+  let filters = [`type: "${types[level]}"`, `str: "${str}"`, first && `first: ${first}`].filter(Boolean);
+  if (parentUuids) {
+    filters.push(`parent_Uuid_In: ["${parentUuids.join('", "')}"]`);
+  }
+  let projections = ["id", "uuid", "type", "code", "name", nestParentsProjections(level)];
```

Comment:
I think 'projections' and 'payload' are not updated in this context therefore we can use const instead of let @olewandowski1 .
filters is ok, because there is 'push'.